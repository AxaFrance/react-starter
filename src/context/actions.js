export const GET_TITLE = 'GET_TITLE';
