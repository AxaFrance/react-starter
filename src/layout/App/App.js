import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import {
  AuthenticationProvider,
  InMemoryWebStorage,
  OidcSecure,
} from '@axa-fr/react-oidc-context';

import Header from 'layout/Header';
import Routes from 'layout/Routes';
import Footer from 'layout/Footer';
import oidcConfiguration from 'configuration';

import './App.scss';

const App = ({ oidcEnabled = true }) => {
  return (
    <AuthenticationProvider
      configuration={oidcConfiguration.is4}
      UserStore={InMemoryWebStorage}
      isEnabled={oidcEnabled}>
      <Router>
        <OidcSecure>
          <Header />
          <div className="app">
            <Routes />
          </div>
          <Footer />
        </OidcSecure>
      </Router>
    </AuthenticationProvider>
  );
};

export default App;
