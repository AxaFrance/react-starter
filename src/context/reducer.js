import { GET_TITLE } from './actions';

const reducer = (state, { type, payload }) => {
  switch (type) {
    case GET_TITLE: {
      return {
        ...state,
        title: payload,
      };
    }
    default:
      return state;
  }
};

export default reducer;
