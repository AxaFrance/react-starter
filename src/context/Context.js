import React, { useReducer } from 'react';
import reducer from './reducer';

const Context = React.createContext();

const initialState = {
  title: 'Titre Accueil',
};

const ContextProvider = ({ children }) => {
  const [context, dispatch] = useReducer(reducer, initialState);
  const value = { context, dispatch };

  return <Context.Provider value={value}>{children}</Context.Provider>;
};

export { Context, ContextProvider };
