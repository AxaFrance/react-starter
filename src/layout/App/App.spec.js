import React from 'react';
import { render, waitFor } from '@testing-library/react';
import { StaticRouter } from 'react-router-dom';
import { Context } from 'context/Context';

import App from './App';

const wrapper = ({ children }) => (
  <StaticRouter context={{}}>
    <Context.Provider value={{ context: { title: 'Titre Accueil' } }}>
      {children}
    </Context.Provider>
  </StaticRouter>
);

it('1. Renders Menu component without crashing', async () => {
  const { asFragment } = render(<App oidcEnabled={false} />, {
    wrapper,
  });

  await waitFor(() => expect(asFragment()).toMatchSnapshot());
});
