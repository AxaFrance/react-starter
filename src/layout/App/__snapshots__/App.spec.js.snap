// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. Renders Menu component without crashing 1`] = `
<DocumentFragment>
  <div
    class="af-header"
  >
    <div
      class="container af-header__wrapper"
    >
      <header
        class="af-header__content"
        role="banner"
      >
        <div
          class="af-header__name"
        >
          <a
            class="af-logo"
            href="/"
          >
            <img
              alt="Logo"
              class="af-logo__brand"
              src="logo-axa.svg"
            />
          </a>
          <h2
            class="af-header__title"
          >
            Starter kit
            <small
              class="af-header__subtitle"
            >
              A little project as a demo for all our tools
            </small>
          </h2>
        </div>
        <div
          class="af-info-user"
        >
          <a
            class="af-info-user__link"
            href="#"
            title="Voir mon profil"
          >
            <span
              class="af-info-user__name"
            >
              Utilisateur non connecté
            </span>
            <span
              class="af-info-user__profile"
            >
              [Inspecteur]
            </span>
          </a>
        </div>
      </header>
    </div>
  </div>
  <div
    class="app"
  >
    <div
      class="af-nav-container"
    >
      <button
        aria-controls="mainmenu"
        aria-haspopup="true"
        class="mask fade show"
        id="togglemenu"
        type="button"
      />
      <nav
        aria-label="Menu principal"
        class="af-nav af-drawer left show"
        role="navigation"
      >
        <div
          class="container af-nav__wrapper"
        >
          <div
            class="af-nav__aside-header"
          >
            <h4
              class="af-nav__aside-title"
            >
              Menu
            </h4>
            <button
              aria-controls="mainmenu"
              aria-haspopup="true"
              class="af-nav__aside-close"
              id="togglemenu"
              type="button"
            >
              ×
            </button>
          </div>
          <ul
            aria-labelledby="togglemenu"
            class="af-nav__list "
            id="mainmenu"
            role="menubar"
          >
            <li
              class="af-nav__item af-nav__item--active "
              role="none"
            >
              <a
                aria-expanded="false"
                class="af-nav__link"
                href="/"
                role="menuitem"
                tabindex="-1"
              >
                Accueil
              </a>
            </li>
            <li
              class="af-nav__item "
              role="none"
            >
              <a
                aria-expanded="false"
                class="af-nav__link"
                href="/dashboard"
                role="menuitem"
                tabindex="-1"
              >
                Dashboard
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
    <div
      class="af-title-bar"
    >
      <div
        class="container af-title-bar__wrapper"
      >
        <h1
          class="af-title-bar__title"
        >
          Accueil
        </h1>
      </div>
    </div>
    <div
      class="home container"
    >
      <h1
        class="af-title--content"
      >
        Accueil
      </h1>
      <h2 />
      <p>
        Donec sollicitudin molestie malesuada. Donec sollicitudin molestie malesuada. Donec rutrum congue leo eget malesuada. Vivamus suscipit tortor eget felis porttitor volutpat. Sed porttitor lectus nibh. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Proin eget tortor risus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque in ipsum id orci porta dapibus.
      </p>
      <p>
        Sed porttitor lectus nibh. Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Mauris blandit aliquet elit, eget tincidunt nibh pulvinar a. Sed porttitor lectus nibh. Quisque velit nisi, pretium ut lacinia in, elementum id enim. Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Nulla quis lorem ut libero malesuada feugiat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus suscipit tortor eget felis porttitor volutpat. Sed porttitor lectus nibh.
      </p>
    </div>
  </div>
  <footer
    class="footer"
  >
    <div
      class="container-fluid container"
    >
      <div
        class="af-logo"
      >
        <img
          alt="Logo Axa"
          class="af-logo__brand"
          src="logo-axa.svg"
        />
      </div>
      <div
        class="af-footer-content"
      >
        <div
          class="footer__content"
        >
          <div
            class="footer__base"
          >
            Titre du footer
            <i>
              Tous droits réservés
            </i>
          </div>
          <div
            class="footer__info"
          >
            <p>
              Version: 0.1.0
            </p>
          </div>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
